### PR TITLE
Allow native with cores and nodes.

### PR DESCRIPTION
**Pull request by christinaholtNOAA**
**Synopsis**

The Rocoto native flag may be used to specify resources if no nodes/cores flags are available, but it may also be used alongside one of those for any variety of reasons. When the bug was found, the reservation flag was the culprit for finding the bug.

**Type**

- [x] Bug fix (corrects a known issue)
- [ ] Code maintenance (refactoring, etc. without behavior change)
- [ ] Documentation
- [ ] Enhancement (adds new functionality)
- [ ] Tooling (CI, code-quality, packaging, revision-control, etc.)

**Impact**

- [ ] This is a breaking change (changes existing functionality)
- [x] This is a non-breaking change (existing functionality continues to work as expected)

**Checklist**

- [x] I have added myself and any co-authors to the PR's _Assignees_ list.
- [x] I have reviewed the documentation and have made any updates necessitated by this change.

### CODE REVIEW

File: src/uwtools/resources/jsonschema/rocoto.jsonschema [hunk 1]
```diff
@@ -290,6 +290,23 @@
...
+          "required": [
+            "native"
+          ]
+        },
+        {
+          "required": [
+            "nodes"
+          ]
+        }
+      ],
```
Comment (christinaholtNOAA) [inline src/uwtools/resources/jsonschema/rocoto.jsonschema:309]: I assume the formatter did this move when it was alphabetized?

File: src/uwtools/tests/test_schemas.py [hunk 2]
```diff
@@ -1664,6 +1664,18 @@ def test_schema_rocoto_workflow_cycledef():
...
+def test_schena_rocoto_task_resources():
+    errors = schema_validator("rocoto", "$defs", "task", "properties")
+    # Basic resource options
+    assert not errors([{"cores": 1}])
+    assert not errors([{"native": "abc"}])
+    assert not errors([{"native": {"cyclestr": {"value": "def"}}}])
+    assert not errors([{"nodes": "1:ppn=12"}])
+    # Combined valid resources
+    assert not errors([{"cores": 1, "native": "abc"}])
+    assert not errors([{"nodes": "1:ppn=12", "native": "abc"}])
```
Comment (christinaholtNOAA) [inline src/uwtools/tests/test_schemas.py:1676]: I really wanted there to be an extra "don't allow cores and nodes" rule but do not have time to work out the logic with this bug fix. Maybe in a future PR?

Comment (NaureenBharwaniNOAA) [inline src/uwtools/tests/test_schemas.py:1676]:
```suggestion
    assert not errors([{"native": "abc", "nodes": "1:ppn=12"}])
```

I reordered these so they were in alphabetical ordering.

Comment (maddenp-noaa) [reply src/uwtools/tests/test_schemas.py:1676]: Do you mean that we'd like to enforce that `cores` and `nodes` are mutually exclusive? I think that can be done in the schema and tested here. If you open a Jira ticket for it, it'd be a nice fill-in task.

**Review by maddenp-noaa — APPROVED** [review]

**Review by WeirAE — APPROVED** [review]

File: src/uwtools/tests/test_schemas.py (hunk 2 above)
Comment (christinaholtNOAA) [reply src/uwtools/tests/test_schemas.py]: I think that is how we should go, yes. @WeirAE Does your 👍 mean that you're creating a Jira ticket?

**Pull request merged**